lindi/LindiH5pyFile/LindiH5pyFile.py

```diff
@@ -52,6 +52,9 @@ def from_lindi_file(url_or_path: str, *, mode: LindiFileMode = "r", staging_area
 
         For a description of parameters, see from_reference_file_system().
         """
+        if local_file_path is None:
+            if not url_or_path.startswith("http://") and not url_or_path.startswith("https://"):
+                local_file_path = url_or_path
```

Comment:
Set the local file path (the one that can get edited) to be equal to the url_or_path in the case that it is not provided and is not a url